### PR TITLE
#93 Update readme and doc script

### PR DESCRIPTION
**Pull request by vbshuliar**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build pharmaverseadam site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new datasets occur on the "[Reference](https://pharmaverse.github.io/pharmaverseadam/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately 
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by vbshuliar** [comment]
The create_adams_data.R script has to be rerun in order to generate new scripts with the updated documentation. I can't do it from my PC as GitHub puts a limit on the amount of API calls.

File: README.md [hunk 1]
```diff
@@ -1,45 +1,53 @@
...
-action can be triggered manually by the package maintainers in two
-scenarios:
+The ADaM contents of this package is populated by an action that executes the `{admiral}`, `{admiralonco}`, `{admiralophtha}`, `{admiralvaccine}` and `{admiralpeds}` templates and saves the resulting datasets here. This action can be triggered manually by the package maintainers in two scenarios:
 
--   Regularly, upon new releases of `{admiral}`, `{admiralonco}`,
-    `{admiralophtha}` and `{admiralvaccine}`;
--   Ad-hoc, whenever templates in the above packages have been updated
-    but releases are far away in the calendar. In this case, the ADaM
-    datasets are created using the development version of the templates.
+-   Regularly, upon new releases of `{admiral}`, `{admiralonco}`, `{admiralophtha}` and `{admiralvaccine}`;
```
Comment (manciniedoardo) [inline README.md:20]:
```suggestion
-   Regularly, upon new releases of these packages;
```

File: README.md [hunk 2]
```diff
@@ -1,45 +1,53 @@
...
+
+1.  **Metadata Preparation**:
+    -   Ensure `adams-specs.xlsx` contains up-to-date metadata for all datasets.
+    -   Missing fields default to "No label/description/source available."
+2.  **Run Script**:
+    -   The `data-raw/create_adams_data.R` script dynamically retrieves metadata, enriches it with dataset attributes, and generates `.R` files in the `R/` directory.
+3.  **Generate Documentation**:
+    -   Use `roxygen2::roxygenize()` to update `.Rd` files in the `man/` directory.
+
+## Key Benefits of this Documentation Approach
```
Comment (manciniedoardo) [inline README.md:47]: Remove this section - unneeded for README

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Great work! Left a couple of comments.

**Comment by manciniedoardo** [comment]
@vbshuliar in addition to my above review can you please also restructure the reference page so that it talks about datasets and not functions? thanks
